x-pack/legacy/plugins/apm/public/components/shared/LocalUIFilters/Filter/index.tsx

```diff
@@ -0,0 +1,168 @@
+  value,
+  showCount
+}: Props) => {
+  const [showPopover, setShowPopover] = useState(false);
+
+  const toggleShowPopover = () => setShowPopover(show => !show);
+
+  const button = (
+    <FilterTitleButton onClick={toggleShowPopover}>{title}</FilterTitleButton>
+  );
+
+  const items: Option[] = useMemo(
+    () =>
+      options.map(option => ({
+        label: option.name,
+        append: showCount ? (
+          <Counter>
+            <EuiText size="xs">{option.count}</EuiText>
+          </Counter>
+        ) : null,
+        checked: value.includes(option.name) ? 'on' : undefined
+      })),
+    [value, options, showCount]
+  );
+
+  const [visibleOptions, setVisibleOptions] = useState(items);
+
+  useEffect(() => {
+    setVisibleOptions(items);
+  }, [items]);
+
+  return (
+    <>
+      <Popover
+        id={`local-filter-${name}`}
+        closePopover={toggleShowPopover}
+        button={button}
+        isOpen={showPopover}
+        panelPaddingSize="s"
+        anchorPosition="downLeft"
+      >
+        <EuiSelectable
+          onChange={selectedOptions => {
+            setVisibleOptions(selectedOptions);
+          }}
+          options={visibleOptions}
+          searchable={true}
+        >
+          {(list, search) => (
+            <SelectContainer>
+              <EuiFlexGroup direction="column" gutterSize="none">
+                <EuiFlexItem grow={true}>
+                  <EuiTitle size="xxxs" textTransform="uppercase">
+                    <h4>
+                      {i18n.translate('xpack.apm.applyFilter', {
+                        defaultMessage: 'Apply {title} filter',
+                        values: { title }
+                      })}
+                    </h4>
+                  </EuiTitle>
+                  <EuiHorizontalRule margin="none" />
+                  <EuiSpacer size="s" />
+                  {search}
+                  <EuiSpacer size="s" />
+                  {list}
+                  <EuiSpacer size="s" />
+                </EuiFlexItem>
+                <EuiFlexItem grow={false}>
+                  <ApplyButton
+                    color="primary"
+                    fill={true}
+                    onClick={() => {
+                      const newValue = visibleOptions
+                        .filter(option => option.checked === 'on')
+                        .map(option => option.label);
+
+                      setShowPopover(false);
+                      onChange(newValue);
+                    }}
+                    size="s"
+                  >
+                    {i18n.translate('xpack.apm.applyOptions', {
+                      defaultMessage: 'Apply options'
+                    })}
+                  </ApplyButton>
+                </EuiFlexItem>
+              </EuiFlexGroup>
+            </SelectContainer>
+          )}
+        </EuiSelectable>
+      </Popover>
+      {value.length ? (
+        <>
+          <FilterBadgeList onChange={onChange} value={value} />
```

Comment:
Should this be called `onRemove` or something like it (can actually remember what exactly it does)